src/frontend/src/pages/GanttPage/GanttChartComponents/GanttTaskBar/GanttTaskBarEdit.tsx

```diff
@@ -85,169 +94,216 @@ const GanttTaskBarEdit = ({
     e.preventDefault();
   };
   const onDrop = (day: Date) => {
-    const days = roundToMultipleOf7(differenceInDays(day, event.start));
-    createChange({ id, eventId: event.id, type: 'shift-by-days', days });
+    const days = roundToMultipleOf7(differenceInDays(day, task.start));
+    createChange({ id, element: transformGanttTaskToWorkPackage(task), type: 'shift-by-days', days });
   };
 
   useEffect(() => {
     if (bounds.width !== 0 && width === 0) {
       setWidth(bounds.width);
     }
-  }, [bounds, event.end, event.start, width]);
+  }, [bounds, width]);
 
   const [showAddWorkPackageModal, setShowAddWorkPackageModal] = useState(false);
 
-  return (
-    <div
-      style={{ position: 'relative', width: '100%', marginTop: 10 }}
-      onMouseUp={handleMouseUp}
-      onMouseMove={handleMouseMove}
-    >
-      <AddWorkPackageModal
-        showModal={showAddWorkPackageModal}
-        handleClose={() => setShowAddWorkPackageModal(false)}
-        addWorkPackage={(workPackage) => {
-          const dup = id + event.workPackages.length + 1;
-          addWorkPackage({
-            id: dup,
-            name: workPackage.name,
-            start: new Date(),
-            end: new Date(Date.now() + 1000 * 60 * 60 * 24 * 7),
-            type: 'task',
-            workPackages: [],
-            projectId: event.id,
-            projectNumber: event.projectNumber,
-            carNumber: event.carNumber,
-            stage: workPackage.stage,
-            styles: {
-              color: GanttWorkPackageTextColorPipe(workPackage.stage),
-              backgroundColor: GanttWorkPackageStageColorPipe(workPackage.stage, WbsElementStatus.Inactive)
-            }
-          });
+  const createChangeHandler = (change: GanttChange) => {
```

Comment:
I think this handler is redundant, you should be able to just use createChange

Reply:
i think this is leftover from something i was experimenting with